src/Courses/CourseDetailsTabs/CourseUsersList.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import { ListGroup, ListGroupItem, Row, Col, Form, FormControl, Button } from 'react-bootstrap';
+import { UserObject } from '../CourseInterfaces';
+import _ from 'lodash';
+
+interface CourseUsersListProps {
+    users: Array<UserObject>;
+    activeUsers: Array<Set<number>>;
+    setActive: React.Dispatch<React.SetStateAction<Set<number>[]>>;
+}
+
+export const CourseUsersList: React.FC<CourseUsersListProps> = ({users, activeUsers, setActive}) => {
+    // TODO: This should be refactored into a reusable TwoState button.
+    const [selectAllState, setSelectAllState] = useState(false);
+
+    // Add or remove a user's ID to the active users array.
+    const onClickStudent = (id: number) => {
+        if (activeUsers[0].has(id)) {
```

Comment:
If so why ^

Reply:
This is a bit of a hack to use a Set, which provides implicit uniqueness, within state. Because the Set is an object, state won't update when it updates because the reference won't change. Creating an array each time with a set inside it will cause the state to update.

Reply:
I would rather we had an object that was described in an interface with a data property of type any and then update that, I think activeUsers.data is easier to read then activeUsers[0]